Sources/System/FileControlRaw.swift

```diff
@@ -0,0 +1,726 @@
+/*
+ This source file is part of the Swift System open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift System project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ */
+
+#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+import Darwin
+#elseif os(Linux) || os(FreeBSD) || os(Android)
+import CSystem
+import Glibc
+#elseif os(Windows)
+import CSystem
+import ucrt
+#else
+#error("Unsupported Platform")
+#endif
+
+// RawRepresentable wrappers
+extension FileDescriptor {
+  /// File descriptor flags.
+  @frozen
+  public struct Flags: OptionSet {
+    @_alwaysEmitIntoClient
+    public let rawValue: CInt
+
+    @_alwaysEmitIntoClient
+    public init(rawValue: CInt) { self.rawValue = rawValue }
+
+    /// The given file descriptor will be automatically closed in the
+    /// successor process image when one of the execv(2) or posix_spawn(2)
+    /// family of system calls is invoked.
+    ///
+    /// The corresponding C global is `FD_CLOEXEC`.
+    @_alwaysEmitIntoClient
+    public static var closeOnExec: Flags { Flags(rawValue: FD_CLOEXEC) }
+  }
+
+  /// File status flags.
+  @frozen
+  public struct StatusFlags: OptionSet {
+    @_alwaysEmitIntoClient
+    public let rawValue: CInt
+
+    @_alwaysEmitIntoClient
+    public init(rawValue: CInt) { self.rawValue = rawValue }
+
+    @_alwaysEmitIntoClient
+    fileprivate init(_ raw: CInt) { self.init(rawValue: raw) }
+
+    /// Non-blocking I/O; if no data is available to a read
+    /// call, or if a write operation would block, the read or
+    /// write call throws `Errno.resourceTemporarilyUnavailable`.
+    ///
+    /// The corresponding C constant is `O_NONBLOCK`.
+    @_alwaysEmitIntoClient
+    public static var nonblocking: StatusFlags { StatusFlags(O_NONBLOCK) }
+
+    /// Force each write to append at the end of file; corre-
+    /// sponds to `OpenOptions.append`.
+    ///
+    /// The corresponding C constant is `O_APPEND`.
+    @_alwaysEmitIntoClient
+    public static var append: StatusFlags { StatusFlags(O_APPEND) }
+
+    /// Enable the SIGIO signal to be sent to the process
+    /// group when I/O is possible, e.g., upon availability of
+    /// data to be read.
+    ///
+    /// The corresponding C constant is `O_ASYNC`.
+    @_alwaysEmitIntoClient
+    public static var async: StatusFlags { StatusFlags(O_ASYNC) }
+  }
+
+  /// Advisory record locks.
+  ///
+  /// The corresponding C type is `flock`.
+  @frozen
+  public struct FileLock: RawRepresentable {
+    @_alwaysEmitIntoClient
+    public var rawValue: CInterop.FileLock
+
+    @_alwaysEmitIntoClient
+    public init(rawValue: CInterop.FileLock) { self.rawValue = rawValue }
+
+    /// The type of the lock.
+    @frozen
+    public struct LockType: RawRepresentable, Hashable {
```

Comment:
I really try to avoid having a `.Type` that isn't the metatype of a generic type parameter, but I don't know how confusing this would end up being in practice.

Reply:
Per man page guidance, I'm removing this in favor of `flock(2)` support. There may still be the raw commands / constants if someone really wants it, but no need for these types.